engine/gameobject/src/gameobject/test/component/test_gameobject_component.cpp

```diff
@@ -615,6 +615,106 @@ TEST_F(ComponentTest, FinalCallsFinal)
     dmGameObject::PostUpdate(m_Register);
 }
 
+
+struct ComponentApiTestContext
+{
+    int m_Created;
+    int m_Destroyed;
+    void* m_CreateContext;
+    void* m_DestroyContext;
+} g_ComponentApiTestContext;
+
+static dmResource::Result ResourceTypeTestResourceCreate(const dmResource::ResourceCreateParams& params)
+{
+    params.m_Resource->m_Resource = malloc(1);
+    params.m_Resource->m_ResourceSize = 1;
+    return dmResource::RESULT_OK;
+}
+
+static dmResource::Result ResourceTypeTestResourceDestroy(const dmResource::ResourceDestroyParams& params)
+{
+    free(params.m_Resource->m_Resource);
+    return dmResource::RESULT_OK;
+}
+
+static dmGameObject::Result ComponentTypeTest_Create(const dmGameObject::ComponentTypeCreateCtx* ctx, dmGameObject::ComponentType* type)
+{
+    g_ComponentApiTestContext.m_Created = 1;
+    g_ComponentApiTestContext.m_CreateContext = malloc(1);
+
+    ComponentTypeSetContext(type, g_ComponentApiTestContext.m_CreateContext);
+    return dmGameObject::RESULT_OK;
+}
+
+static dmGameObject::Result ComponentTypeTest_Destroy(const dmGameObject::ComponentTypeCreateCtx* ctx, dmGameObject::ComponentType* type)
+{
+    g_ComponentApiTestContext.m_Destroyed = 1;
+    g_ComponentApiTestContext.m_DestroyContext = ComponentTypeGetContext(type);
+
+    ComponentTypeSetContext(type, g_ComponentApiTestContext.m_CreateContext);
```

Comment:
Normally, you'd delete the context here (the point of this PR), but we keep it in order to check that the context is stored/retrieved properly.